tools/optimizer/simple_ast.cpp

```diff
@@ -93,7 +93,9 @@ struct StackedStack { // a stack, on the stack
         memcpy(storage, old, sizeof(T)*used);
         alloced = true;
       } else {
-        storage = (T*)realloc(storage, sizeof(T)*available);
+        T *newStorage = (T*)realloc(storage, sizeof(T)*available);
+        assert(newStorage || !storage);
```

Comment:
No, if I just do `assert(storage);` on this line, the Visual Studio static code analyzer will still issue the warning about a memory leak occurring. Should I just do `assert(newStorage);` instead?